app/src/views/home/home_screen.tsx

```diff
@@ -273,6 +274,16 @@ export default function HomeScreen({
             navigateToCamera();
           }
         }}
+        rightAction={() => {
+          if (userIsGuest) {
+            setShowGuestPopup("Camera");
+          } else {
+            navigation.navigate("Memories", {
+              navigation,
+              user: user,
+            });
+          }
+        }}
```

Comment:
This is cool ! I'm sure the icon could be adapted to match the goal and be more intuitive